src/components/actions/extendApplicationRoute.ts

```diff
@@ -0,0 +1,63 @@
+import { combineRequestParams } from '../utilityFunctions'
+import { PermissionPolicyType, Trigger } from '../../generated/graphql'
+import { DateTime } from 'luxon'
+import DBConnect from '../databaseConnect'
+import { getPermissionNamesFromJWT } from '../data_display/helpers'
+
+export const routeExtendApplication = async (request: any, reply: any) => {
+  const { applicationId, eventCode, extensionTime, data } = combineRequestParams(request, 'camel')
+
+  // Check permissions first -- currently we just check user has ANY Reviewing
+  // permissions for current template.
+  // TO-DO: define specific permission names required for extending deadlines
```

Comment:
I would quite like to add this, as I think we'd want more specificity over who can extend. Will see how much time we have in the Angola budget when we get further along.